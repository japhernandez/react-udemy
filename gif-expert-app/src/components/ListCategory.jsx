import React from "react";
import PropTypes from "prop-types";
import GifItem from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

const ListCategory = ({ category }) => {
  const { loading, data: images } = useFetchGifs(category);

  return (
    <>
      {loading ? (
        <p>Cargando...</p>
      ) : (
        images.map((img) => <GifItem key={img.id} {...img} />)
      )}
    </>
  );
};

ListCategory.propTypes = {
  category: PropTypes.string.isRequired,
};

export default ListCategory;
