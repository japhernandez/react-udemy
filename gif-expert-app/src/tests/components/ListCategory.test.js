import React from "react";
import { shallow } from "enzyme";
import ListCategory from "../../components/ListCategory";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Component <ListCategory />", () => {
  const category = "One Punch";

  it("should must show component correctly", () => {
    useFetchGifs.mockReturnValue({ data: [], loading: true });
    const wrapper = shallow(<ListCategory category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should show items when the useFetchGifs data is loaded", () => {
    const gifs = [
      {
        id: "123490",
        url: "https://localhost:6000/image.jpg",
        title: "Imagen de muestra",
      },
    ];
    useFetchGifs.mockReturnValue({ data: gifs, loading: false });
    const wrapper = shallow(<ListCategory category={category} />);
    expect(wrapper).toMatchSnapshot();
    // 1. El componente <GifItem /> debe de existir
    expect(wrapper.find("GifItem").length).toBe(gifs.length);
    // 2. El parrajo no debe de existir por que el <GifItem /> tiene elementos
    expect(wrapper.find("p").exists()).toBe(false);
  });
});
