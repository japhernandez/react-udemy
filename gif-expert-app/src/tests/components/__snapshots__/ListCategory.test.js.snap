// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <ListCategory /> should must show component correctly 1`] = `
<Fragment>
  <p>
    Cargando...
  </p>
</Fragment>
`;

exports[`Component <ListCategory /> should show items when the useFetchGifs data is loaded 1`] = `
<Fragment>
  <GifItem
    id="123490"
    key="123490"
    title="Imagen de muestra"
    url="https://localhost:6000/image.jpg"
  />
</Fragment>
`;
